index.src.html

```diff
@@ -2958,80 +3017,232 @@
+        2. If either of |optRect|.{{VideoFrameRect/width}} or
+            {{VideoFrameRect/height}} is `0`, return a {{TypeError}}.
+        3. If the sum of |optRect|.{{VideoFrameRect/left}} and
+            |optRect|.{{VideoFrameRect/width}} is greater than
+            {{VideoFrame/[[coded width]]}}, return a {{TypeError}}.
+        4. If the sum of |optRect|.{{VideoFrameRect/top}} and
+            |optRect|.{{VideoFrameRect/height}} is greater than
+            {{VideoFrame/[[coded height]]}}, return a {{TypeError}}.
+        5. Assign |optRect| to |sourceRect|.
+    3. Let |planeIndex| be `0`.
+    4. Let |numPlanes| be the number of planes as defined by
+        {{VideoFrame/[[format]]}}
+    5. While |planeIndex| is less than |numPlanes|:
+
+        NOTE: The following steps validate |rect| is sample-aligned for this
+            frame's {{VideoFrame/[[format]]}}.
+
+        1. Let |plane| be the Plane identified by |planeIndex| as defined by
+            {{VideoFrame/[[format]]}}.
+        2. Let |sampleWidth| be the horizontal pixel size of each subsample
+            for |plane|.
+        3. Let |sampleHeight| be the vertical pixel size of each subsample
+            for |plane|.
+        4. If |sourceRect|.{{VideoFrameRect/left}} and
+            |sourceRect|.{{VideoFrameRect/width}} cannot be evenly divided by
+            |sampleWidth|, throw a {{TypeError}}.
+        5. If |sourceRect|.{{VideoFrameRect/top}} and
+            |sourceRect|.{{VideoFrameRect/height}} cannot be evenly divided by
+            |sampleHeight|, throw a {{TypeError}}.
+        6. Increment |planeIndex| by `1`
+    6. Return |sourceRect|.
+
+: <dfn for=VideoFrame>Compute Layout and Allocation Size</dfn> (with
+    |parsedRect| and |options|)
+:: 1. Let |numPlanes| be the number of planes as defined by
+        {{VideoFrame/[[format]]}}
+    2. If {{VideoFrameCopyToOptions/layout}} [=map/exists=] its size does not
+        equal |numPlanes|, throw a {{TypeError}}.
+    3. Let |minAllocationSize| be `0`.
+    4. Let |computedLayouts| be a new [=list=].
+    5. Let |endOffsets| be a new [=list=].
+    6. Let |planeIndex| be `0`.
+    7. While |planeIndex| < |numPlanes|:
+        1. Let |plane| be the Plane identified by |planeIndex| as defined by
+            {{VideoFrame/[[format]]}}.
+        2. Let |sampleBytes| be the number of bytes per sample for |plane|.
+        3. Let |sampleWidth| be the horizontal pixel size of each subsample
+            for |plane|.
+        4. Let |sampleHeight| be the vertical pixel size of each subsample
+            for |plane|.
+        5. Let |sampleWidthBytes| be the product of multiplying |sampleWidth| by
+            |sampleBytes|.
+        6. Let |computedLayout| be a new [=computed plane layout=].
+        7. Assign quotient of dividing |parsedRect|.{{VideoFrameRect/top}} by
+            |sampleHeight| to |computedLayout|'s [=computed plane layout/sourceTop=].
```

Comment:
`Set sourceTop to the result of the integer division of top by sampleHeight` read better imo.

Reply:
Done, here and below.